test/extern/fnPtrs/storeFnsThenPass.chpl

```diff
@@ -0,0 +1,18 @@
+require 'externCFns.h';
+
+proc foo() {
+  writeln("Hello!");
+}
+
+extern proc bar(f);
+
+proc baz(f) {
+  f();
+}
+
+var x = c_fnPtrTo(foo);
+var y = foo;
+  
+bar(x);
+baz(y);
+ 
```

Comment:
Related to the test itself, did you try reusing the stored version of the function?